x/validators/handler.go

```diff
@@ -0,0 +1,94 @@
+package validators
+
+import (
+	"github.com/confio/weave"
+	"github.com/confio/weave/errors"
+	"github.com/confio/weave/x"
+)
+
+type AuthCheckAddress = func(auth x.Authenticator, ctx weave.Context) CheckAddress
```

Comment:
pp: the interface should be declared where it is used. in this case in the controller.

Reply:
agreed

Reply:
the idea here is that controller does not know about context, so we define the type where we pass it. AuthCheckAddress is only used here, CheckAddress I moved to the controller.